### PR TITLE
Homework "Narzędzia #2: git, vim, kompilatory"

### PR DESCRIPTION
**Pull request by KGebski0036**

### CODE REVIEW

File: src/functions.cpp [hunk 1]
```diff
@@ -1,26 +1,20 @@
-#include "../inc/functions.hpp"    // TODO: add proper compilation flag and change to #include "functions.hpp"
-#include <algorithm>
+#include "functions.hpp"
```
Comment (DamiXioR) [inline src/functions.cpp:1]: in my opinion you removed min function

File: src/functions.cpp [hunk 2]
```diff
@@ -1,26 +1,20 @@
...
 #include <iostream>
-#include <limits>
-
+#include <algorithm>
+    
 void print(std::vector<int> const & numbers) {
-    for (int i = 0; i < numbers.size(); ++i) {
+    for (size_t i = 0; i < numbers.size(); ++i) {
```
Comment (DamiXioR) [inline src/functions.cpp:6]: looks better if you use size_t with std::

File: src/main.cpp [hunk 3]
```diff
@@ -1,14 +1,14 @@
 #include <vector>
 #include <iostream>
-#include "../inc/functions.hpp"    // TODO: add proper compilation flag and change to #include "functions.hpp"
+#include "functions.hpp"    // TODO: add proper compilation flag and change to #include "functions.hpp"
```
Comment (DamiXioR) [inline src/main.cpp]: remove comments

File: Makefile [hunk 4]
```diff
@@ -1,3 +1,3 @@
 all: src/*.cpp inc/*.hpp
-	g++ -std=c++17 src/*.cpp -o main
+	g++ -std=c++17 src/*.cpp -Wall -Iinc -Wextra -pedantic -Werror -Wconversion -O3 -g -o main
```
Comment (DamiXioR) [inline Makefile:2]: -I./inc looks better

**Review by DamiXioR — COMMENTED** [review]
some ideas

**Review by Marcinati — CHANGES_REQUESTED** [review]
1. Files name should be a "tutor" instead of Test.
2. Removing any functionality of program (here: minimum) was forbidden, you should restore this.

File: inc/functions.hpp [hunk 5]
```diff
@@ -2,5 +2,5 @@
 
 void print(std::vector<int> const &);
 void sort(std::vector<int> &);
-int min(std::vector<int> const &);
```
Comment (ziobron) [inline inc/functions.hpp:5]: You shouldn't remove min function.

File: Test [hunk 6]
```diff
@@ -0,0 +1,1012 @@
+===============================================================================
+=    W i t a j   w   t u t o r i a l u   V I M - a      -    Wersja  1.7.     =
+===============================================================================
+<<<<<<< HEAD
+
+=======
+ 
+>>>>>>> vimtutor
```
Comment (ziobron) [inline Test]: Your merge was not successful. Please fix it. If you need help let us know on Discord.
+ the name of this file should be tutor.

**Review by ziobron — CHANGES_REQUESTED** [review]
Judging from the commit names you have not done merges properly. They are named "marge" (with typo), so probably it was your own message. Please try doing it once again, maybe from a clean repo. You can just use the existing tutor file, just copy it to a new instance of this repo.

File: src/functions.cpp [hunk 7]
```diff
@@ -1,26 +1,20 @@
...
 }
-
-int min(std::vector<int> const & values) {
-    int minimum = std::numeric_limits<int>::max();
-    for (const auto & value : values) {
-        if (value < minimum) {
-            minimum = value;
-        }
-    }
-    return minimum;
```
Comment (ziobron) [inline src/functions.cpp]: min function and it's usage is still removed. Please bring this function back.

File: src/main.cpp [hunk 8]
```diff
@@ -1,14 +1,14 @@
...
 
 int main() {
     std::vector<int> numbers = {142, 41, 42, 13, 0, -14, 5};
     print(numbers);
     sort(numbers);
     print(numbers);
-    auto minimum = min(numbers);
-    std::cout << "Minimum: " << minimum << '\n';
```
Comment (ziobron) [inline src/main.cpp]: Bring back usage of min

**Review by ziobron — CHANGES_REQUESTED** [review]

File: src/functions.cpp [hunk 9]
```diff
@@ -12,9 +12,14 @@ void print(std::vector<int> const & numbers) {
 void sort(std::vector<int> & unsorted) {
     std::sort(begin(unsorted), end(unsorted));
 }
-   
-void reverse(std::vector<int> & data) {
-    std::reverse(begin(data), end(data));
-}
-
```
Comment (ziobron) [inline src/functions.cpp]: But now reverse function is gone! We want them both :)

**Review by ziobron — CHANGES_REQUESTED** [review]
Now reverse function is missing. We want both reverse and min :)

**Review by ziobron — APPROVED** [review]

**Comment by ziobron** [comment]
+4 XP Narzędzia#2 – vimtutor
+5 XP Narzędzia#2 – git - konflikty
+4 XP Narzędzia#2 – flagi kompilacji
+3 XP Narzędzia#2 – git - cherrypick
+8 XP delivery before 07.06.2020

{ "requirement_ids": [1119, 1999, 1119, 2000, 1119, 2001, 1119, 2002], "points": [2, 4, 2, 5, 2, 4, 2, 3] }

**Pull request closed without merging**